sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -388,7 +388,8 @@ case class CatalogTable(
 
   def toLinkedHashMap: mutable.LinkedHashMap[String, String] = {
     val map = new mutable.LinkedHashMap[String, String]()
-    val tableProperties = properties.map(p => p._1 + "=" + p._2).mkString("[", ", ", "]")
+    val tableProperties = properties.toSeq.sortBy(_._1)
+      .map(p => p._1 + "=" + p._2).mkString("[", ", ", "]")
```

Comment:
Should we sort by entire strings? For example, if you have `key0 = b` and `key0 = a`, you may get unstable output either `[key0 = b, key0 = a]` or `[key0 = a, key0 = b]`

Reply:
ah, sorry, we can guarantee uniqueness of keys.